src/components/Map/Map.tsx

```diff
@@ -68,6 +77,29 @@ export default function Map(props: MapProps): JSX.Element {
     setFirstVisible((fv) => fv || visible);
   }, [visible]);
 
+  const mapRefCb = useCallback(
+    (map: MapRef | null) => {
+      if (map !== null) {
+        mapRef.current = map;
+
+        // Load all needed map images here
+        // It is done in the callbackRef to ensure the images are available quickly enough for the map layers to use
+        // See https://github.com/visgl/react-map-gl/issues/1118#issuecomment-1139976172
+        mapImages?.forEach(({ name, url }) => {
+          if (!map.hasImage(name)) {
+            map.loadImage(url, (error, image) => {
```

Comment:
`loadImage` required a callback as an argument, which is the reason for not using `await`